src/web100srv.c

```diff
@@ -2614,12 +2446,57 @@ int main(int argc, char** argv) {
                   case 'S':
                     SysLogFacility = optarg;
                     break;
+                  case 314:
+                    options.tls = 1;
+                    break;
+                  case 315:
+                    private_key_file = optarg;
+                    break;
+                  case 316:
+                    certificate_file = optarg;
+                    break;
                   case '?':
                     short_usage(argv[0], "");
                     break;
                 }
               }
 
+  if (options.tls) {
+    if (private_key_file == NULL || certificate_file == NULL) {
+      short_usage(argv[0], "TLS requires a private key and a certificate");
+    }
+
+    SSL_library_init();
+    OpenSSL_add_all_algorithms();
+    SSL_load_error_strings();
+    ssl_context = SSL_CTX_new(SSLv23_server_method());
+    if (ssl_context == NULL) {
+      ssl_err = ERR_get_error();
+      ERR_error_string_n(ssl_err, ssl_error, sizeof(ssl_error));
+      printf("SSL_CTX_new: %s\n", ssl_error);
```

Comment:
We should be using the logging functions.  This code should pretty much never be called - SSL_CTX_new can only fail in some very odd scenarios (OOM and the like)